### PR TITLE
Update Github Actions jobs

### PR DESCRIPTION
**Pull request by clangenb**
Changes:
* Moved zombienet test to check-pull-request workflow
* added cargo zepter test, which checks correct crate feature propagation
* Removed obsolete try-runtime check in the release workflow
* Run `check-pull-request` checks always, also on the release branches
* Check-migrations tests if the spec-version is increased if we are on a release branch.

### CODE REVIEW

File: .github/workflows/check-pull-request.yml [hunk 1]
```diff
@@ -27,7 +27,6 @@ env:
 
 jobs:
   fmt:
-    if: ${{ !startsWith(github.head_ref, 'release/') }}
```
Comment (clangenb) [inline .github/workflows/check-pull-request.yml:30]: Here, you might disagree with me, but I thought as releases also have changes in the rust file, we should definitely run all tests again. What are your thoughts here @DidacSF?

File: .github/workflows/check-pull-request.yml [hunk 2]
```diff
@@ -84,3 +103,40 @@ jobs:
           key: ${{ runner.os }}-cargo-test-${{ matrix.runtime }}-${{ hashFiles('**/Cargo.lock') }}
           restore-keys: ${{ runner.os }}-cargo-test-${{ matrix.runtime }}-
       - run: cargo test --all-features --all-targets ${{ env.CARGO_ARGS }} -p ${{ matrix.runtime }}
+
+  zombienet-tests:
+    name: Zombienet tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        binary: [ bajun-node ]
```
Comment (clangenb) [inline .github/workflows/check-pull-request.yml:106]: Note the renaming, but I think this is correct with the new repo structure: https://github.com/ajuna-network/bajun-parachain/pull/26/commits/2207be7bcffef79f5ea8d34f8317e49b387d78c4

File: .github/workflows/check-pull-request.yml [hunk 3]
```diff
@@ -21,13 +21,10 @@ concurrency:
   cancel-in-progress: true
 
 env:
-  CARGO_ARGS: ${{ github.ref == 'refs/heads/develop' && '--release' || '' }}
```
Comment (clangenb) [inline .github/workflows/check-pull-request.yml:24]: I prefer that this is explicit in each run, and this also gave me the freedom to use debug for clippy.

File: .github/workflows/check-pull-request.yml [hunk 4]
```diff
@@ -21,13 +21,10 @@ concurrency:
...
   CARGO_TERM_COLOR: always
-  SKIP_WASM_BUILD: 1
```
Comment (clangenb) [inline .github/workflows/check-pull-request.yml:26]: This should no-longer be global as the zombienet should build wasm - I think the default execution mode is wasm already.

File: .github/workflows/check-pull-request.yml [hunk 5]
```diff
@@ -40,47 +37,106 @@ jobs:
...
+      - name: Install stable Rust
+        uses: actions-rs/toolchain@v1
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-lint-${{ hashFiles('**/Cargo.lock') }}
```
Comment (clangenb) [inline .github/workflows/check-pull-request.yml:59]: I think this corresponds to the default that `rust-cache` does internally, why did you add this?

File: .github/workflows/check-pull-request.yml [hunk 6]
```diff
@@ -40,47 +37,106 @@ jobs:
...
   test:
-    if: ${{ !startsWith(github.head_ref, 'release/') }}
-    name: Test
+    name: Cargo test (all features)
     runs-on: ubuntu-latest
     continue-on-error: false
-    strategy:
-      matrix:
-        runtime:
-          - bajun-runtime
```
Comment (clangenb) [inline .github/workflows/check-pull-request.yml:71]: I removed the runtime from the test, I think we always want to run all tests in the repo anyhow.

File: .github/workflows/check-pull-request.yml [hunk 7]
```diff
@@ -40,47 +37,106 @@ jobs:
...
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-test-${{ matrix.runtime }}-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: ${{ runner.os }}-cargo-test-${{ matrix.runtime }}-
-      - run: cargo test --all-features --all-targets ${{ env.CARGO_ARGS }} -p ${{ matrix.runtime }}
+          # consistent cache across jobs
+          shared-key: "bajun-cache-cargo-release"
```
Comment (clangenb) [inline .github/workflows/check-pull-request.yml:94]: Zombienet and `cargo test --release` share the same cache now

File: .github/workflows/check-pull-request.yml (hunk 1 above)
Comment (DidacSF) [reply .github/workflows/check-pull-request.yml:30]: Yes, I think it's a good idea!

File: .github/workflows/check-pull-request.yml (hunk 5 above)
Comment (DidacSF) [reply .github/workflows/check-pull-request.yml:59]: Honestly, It was added before I even entered Ajuna, never paid much attention to it

**Review by DidacSF — APPROVED** [review]
